File: sorting/insertion_sort/insertion_sort.cpp
#include <vector>
using namespace std; 


vector<int> sort(vector<int> arr){

    for(int i=1; i < arr.size(); i++){
        int temp = arr[i]; 
        int j = i - 1; 
        
        while(j >= 0 && arr[j] > temp){
            arr[j+1] = arr[j]; 
            j--; 
        }
        arr[j+1] = temp; 
    }
    return arr; 
}

File: sorting/insertion_sort/insertion_sort.py
from typing import List


def sort(arr: List) -> List:
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > temp:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = temp

    return arr
